aws/resource_aws_ssm_association.go

```diff
@@ -186,37 +186,67 @@ func resourceAwsSsmAssociationRead(d *schema.ResourceData, meta interface{}) err
 	return nil
 }
 
-func resourceAwsSsmAssocationUpdate(d *schema.ResourceData, meta interface{}) error {
+func resourceAwsSsmAssociationUpdate(d *schema.ResourceData, meta interface{}) error {
 	ssmconn := meta.(*AWSClient).ssmconn
 
-	log.Printf("[DEBUG] SSM association update: %s", d.Id())
+	log.Printf("[DEBUG] SSM Association update: %s", d.Id())
 
 	associationInput := &ssm.UpdateAssociationInput{
 		AssociationId: aws.String(d.Get("association_id").(string)),
 	}
 
+	// AWS creates a new version every time the association is updated, so everything should be passed in the update.
+
+	hasChanges := false
```

Comment:
The `resourceAwsSsmAssociationUpdate` function is not performing any other API calls beyond `UpdateAssociation` so all the `hasChanges` logic can be removed 👍

Reply:
Good point. I'll get rid of this.